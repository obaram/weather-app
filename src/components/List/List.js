import React from 'react';
import {Listitem} from './ListItem/Listitem'
import { Table, Alert} from 'react-bootstrap';
import styles from './List.module.scss'

const List = ({items,handleDelete,unit,error, success}) => (

    <>
    {error.show && <Alert  className={styles.alert} key="alert" variant="warning">{error.msg}</Alert>  }
    {success.show && <Alert  className={styles.alert} key="success" variant="success">{success.msg}</Alert>  } 
   
   <Table striped bordered hover className={styles.wrapper}>
    <thead>
        <tr>
        <th>#</th>
        <th>Miasto</th>
        <th>Åšrednia prognozowana tempretura</th>
        <th></th>
        </tr>
    </thead>
    <tbody>
        
        {
            items.map((item,index) => <Listitem item={item} index={index} handleDelete={handleDelete} key={index} unit={unit}/>)
        }
        
    </tbody>
    </Table>
    </>
)

export default List;
