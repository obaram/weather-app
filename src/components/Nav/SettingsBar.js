import React from 'react';
import { Button } from 'react-bootstrap';
import {Link} from 'react-router-dom';
import styles from './SettingsBar.module.scss'
import FontAwesome from 'react-fontawesome'

const SettingsBar = (props) => (

    <div className={styles.wrapper}>
        <Button variant="outline-primary"><Link to="/settings">
        <FontAwesome
                        className={styles.icon}
                        name='gears'
                        size='1px'
                        className={styles.icon}
                        />                
            Settings</Link></Button>
    </div>

)

export default SettingsBar;