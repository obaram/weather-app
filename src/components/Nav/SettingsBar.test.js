import React from 'react';
import ReactDOM from 'react-dom';
import List from '../List/List';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'; 
import {BrowserRouter,Route, Switch} from 'react-router-dom'
import "../../../setupTest"
import SettingsBar from "./SettingsBar"


  it('renders correctly', () => {
    const tree = renderer.create(<BrowserRouter><SettingsBar /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });


 describe('SettingsBar.js', () => {
     it('should render list item',()=>{     
        const wrapper = shallow(<SettingsBar/>);
        expect(wrapper).toMatchSnapshot();
     })
})
