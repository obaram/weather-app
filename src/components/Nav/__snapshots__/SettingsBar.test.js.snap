// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsBar.js should render list item 1`] = `
<div
  className="wrapper"
>
  <Button
    active={false}
    disabled={false}
    type="button"
    variant="outline-primary"
  >
    <Link
      to="/settings"
    >
      <FontAwesome
        className="icon"
        name="gears"
        size="1px"
      />
      Ustawienia
    </Link>
  </Button>
</div>
`;

exports[`SettingsBar.js should render settings bar 1`] = `
<div
  className="wrapper"
>
  <Button
    active={false}
    disabled={false}
    type="button"
    variant="outline-primary"
  >
    <Link
      to="/settings"
    >
      <FontAwesome
        className="icon"
        name="gears"
        size="1px"
      />
      Ustawienia
    </Link>
  </Button>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="wrapper"
>
  <button
    className="btn btn-outline-primary"
    disabled={false}
    type="button"
  >
    <a
      href="/settings"
      onClick={[Function]}
    >
      <span
        aria-hidden={true}
        className="fa fa-gears fa-1px icon"
      />
      Ustawienia
    </a>
  </button>
</div>
`;
